Core/Layer/Worlds/StatusBar/StatusBarRenderer.cs

```diff
@@ -0,0 +1,1160 @@
+                int lineX = pos.X;
+                
+                if ((alignment & StatusBarAlignment.Right) != 0) lineX -= lineWidth;
+                else if ((alignment & StatusBarAlignment.HCenter) != 0) lineX -= lineWidth / 2;
+                
+                DrawPart(part.Lbl, "CRGREEN", (lineX, cursor.Y));
+                DrawPart(part.Val, comp.Translation, (lineX + part.LblW, cursor.Y));
+                
+                cursor.Y += fontHeight;
+            }
+            else
+            {
+                DrawPart(part.Lbl, "CRGREEN", cursor);
+                cursor.X += part.LblW;
+                
+                DrawPart(part.Val, comp.Translation, cursor);
+                cursor.X += part.ValW + 8; 
+            }
+        }
+    }
+
+    private void DrawChildren(IHudRenderContext hud, StatusBarBaseDef def, Vec2I pos, int containerHeight, StatusBarContext context, float widescreenOffset)
+    {
+        if (def.Children != null)
+        {
+            foreach (var child in def.Children)
+            {
+                DrawElementWrapper(hud, child, pos, containerHeight, context, widescreenOffset);
+            }
+        }
+    }
+
+    private static Vec2I ResolvePosition(StatusBarBaseDef def, Vec2I parentPos, float widescreenOffset)
+    {
+        Vec2I pos = parentPos;
+        pos.X += def.X;
+        pos.Y += def.Y;
+        
+        if (widescreenOffset > 0)
+        {
+            bool dynLeft = (def.Alignment & StatusBarAlignment.DynamicLeft) != 0;
+            bool dynRight = (def.Alignment & StatusBarAlignment.DynamicRight) != 0;
+            int offset = (int)Math.Ceiling(widescreenOffset);
+
+            if (dynLeft && dynRight) { }
+            else if (dynLeft) pos.X -= offset;
+            else if (dynRight) pos.X += offset;
+        }
+        return pos;
+    }
+
+    private static float GetWidescreenOffset(IHudRenderContext hud)
+    {
+        float currentAspect = hud.WindowDimension.AspectRatio;
+        if (currentAspect > Constants.DoomVirtualAspectRatio)
+        {
+            float widthInDoomUnits = 320.0f * (currentAspect / Constants.DoomVirtualAspectRatio);
+            return (widthInDoomUnits - 320.0f) / 2.0f;
+        }
+        return 0f;
+    }
+    
+    private static bool ResolveGlyph(IHudRenderContext hud, string patch, out int width, out int height)
+    {
+        if (hud.Textures.TryGet(patch, out var handle))
+        {
+            width = handle.Dimension.Width;
+            height = handle.Dimension.Height;
+            return true;
+        }
+        else if (hud.Textures.TryGet(patch, out handle, ResourceNamespace.Sprites))
+        {
+            width = handle.Dimension.Width;
+            height = handle.Dimension.Height;
+            return true;
+        }
+        width = 0; 
+        height = 0;
+        return false;
+    }
+    
+    private static string GetHudFontPatch(StatusBarHudFontDef font, char c)
+    {
+        string key = font.Stem + c;
```

Comment:
Here we will have to use another SpanString member variable

m_keySpan.Clear();
m_keySpan.Append(font.Stem);
m_keySpan.Append(c);

var lookup = HudFontPatchCache.GetAlternateLookup<ReadOnlySpan<char>>();
if (lookup.TryGetValue(key, out var cached))
...